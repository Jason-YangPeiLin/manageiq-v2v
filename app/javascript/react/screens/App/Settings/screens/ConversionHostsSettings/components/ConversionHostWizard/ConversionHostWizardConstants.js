export const stepIDs = {
  locationStep: 'conversionHostWizardLocationStep',
  hostsStep: 'conversionHostWizardHostsStep',
  authenticationStep: 'conversionHostWizardAuthenticationStep',
  resultsStep: 'conversionHostWizardResultsStep'
};
