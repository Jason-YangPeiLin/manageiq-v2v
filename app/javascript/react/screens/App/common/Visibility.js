import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'patternfly-react';

// This component has no role on the page, and will be hidden from screen readers (role="presentation").
// Renders the children wrapped in visibility: hidden if the condition is met.
// Either way, the children take up space, visibility will not change the flow of the layout.
// The onKeyUp handler and role are used to satisfy eslint-plugin-jsx-a11y.
const Visibility = ({ children, hidden }) => (
  <span style={{ visibility: hidden ? 'hidden' : '' }} onKeyUp={noop} role="presentation">
    {children}
  </span>
);

Visibility.propTypes = {
  children: PropTypes.node,
  hidden: PropTypes.bool
};

export default Visibility;
