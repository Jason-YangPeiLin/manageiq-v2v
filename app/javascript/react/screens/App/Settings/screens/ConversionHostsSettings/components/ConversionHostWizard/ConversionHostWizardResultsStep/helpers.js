import { OPENSTACK } from '../../../../../../../../../common/constants';
import { VDDK } from '../ConversionHostWizardConstants';

export const getConfigureConversionHostPostBodies = (locationStepValues, hostsStepValues, authStepValues) =>
  hostsStepValues.hosts.map(host => {
    const openstackSpecificProperties =
      locationStepValues.providerType === OPENSTACK ? { auth_user: authStepValues.openstackUser } : {};
    const vmwareAuthProperties =
      authStepValues.transformationMethod === VDDK
        ? { vmware_vddk_package_url: authStepValues.vddkLibraryPath }
        : { vmware_ssh_private_key: authStepValues.vmwareSshKey.body };
    return {
      name: host.name,
      resource_type: host.type,
      resource_id: host.id,
      ssh_key: authStepValues.conversionHostSshKey.body,
      ...openstackSpecificProperties,
      ...vmwareAuthProperties
    };
  });
