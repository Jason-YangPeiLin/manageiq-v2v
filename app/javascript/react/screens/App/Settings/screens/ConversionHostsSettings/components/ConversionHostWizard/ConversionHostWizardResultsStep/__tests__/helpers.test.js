import { RHV, OPENSTACK } from '../../../../../../../../../../common/constants';
import { getConfigureConversionHostPostBodies } from '../helpers';
import { VDDK, SSH } from '../../ConversionHostWizardConstants';

describe('conversion host wizard results step helpers', () => {
  const hostsStepValues = {
    hosts: [{ id: '1', type: 'mock host type', name: 'Mock Host' }]
  };
  const commonAuthStepValues = {
    conversionHostSshKey: { body: 'mock conversion host SSH key body' },
    openstackCaCerts: { body: '' }
  };
  const vddkAuthStepValues = {
    ...commonAuthStepValues,
    transformationMethod: VDDK,
    vddkLibraryPath: 'mock VDDK path'
  };
  const sshAuthStepValues = {
    ...commonAuthStepValues,
    transformationMethod: SSH,
    vmwareSshKey: { body: 'mock vmware SSH key body' }
  };

  describe('for RHV hosts', () => {
    const locationStepValues = { providerType: RHV };

    it('constructs a POST body for a VDDK conversion host', () => {
      const postBodies = getConfigureConversionHostPostBodies(locationStepValues, hostsStepValues, vddkAuthStepValues);
      expect(postBodies).toMatchSnapshot();
    });

    it('constructs a POST body for an SSH conversion host', () => {
      const postBodies = getConfigureConversionHostPostBodies(locationStepValues, hostsStepValues, sshAuthStepValues);
      expect(postBodies).toMatchSnapshot();
    });
  });

  describe('for OSP hosts', () => {
    const locationStepValues = { providerType: OPENSTACK };

    it('constructs a POST body for a VDDK conversion host without CA certs', () => {
      const postBodies = getConfigureConversionHostPostBodies(locationStepValues, hostsStepValues, {
        ...vddkAuthStepValues,
        openstackUser: 'cloud-user'
      });
      expect(postBodies).toMatchSnapshot();
    });

    it('constructs a POST body for an SSH conversion host without CA certs', () => {
      const postBodies = getConfigureConversionHostPostBodies(locationStepValues, hostsStepValues, {
        ...sshAuthStepValues,
        openstackUser: 'cloud-user'
      });
      expect(postBodies).toMatchSnapshot();
    });

    it('constructs a POST body for an SSH conversion host with CA certs', () => {
      const postBodies = getConfigureConversionHostPostBodies(locationStepValues, hostsStepValues, {
        ...sshAuthStepValues,
        openstackUser: 'cloud-user',
        openstackCaCerts: { body: 'mock openstack CA certs body' }
      });
      expect(postBodies).toMatchSnapshot();
    });
  });
});
