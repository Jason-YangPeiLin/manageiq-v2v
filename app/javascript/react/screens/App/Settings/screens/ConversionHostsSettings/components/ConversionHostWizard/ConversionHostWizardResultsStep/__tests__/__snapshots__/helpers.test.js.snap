// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion host wizard results step helpers for OSP hosts constructs a POST body for a VDDK conversion host without CA certs 1`] = `
Array [
  Object {
    "auth_user": "cloud-user",
    "conversion_host_ssh_private_key": "mock conversion host SSH key body",
    "name": "Mock Host",
    "resource_id": "1",
    "resource_type": "mock host type",
    "vmware_vddk_package_url": "mock VDDK path",
  },
]
`;

exports[`conversion host wizard results step helpers for OSP hosts constructs a POST body for an SSH conversion host with CA certs 1`] = `
Array [
  Object {
    "auth_user": "cloud-user",
    "conversion_host_ssh_private_key": "mock conversion host SSH key body",
    "name": "Mock Host",
    "openstack_tls_ca_certs": "mock openstack CA certs body",
    "resource_id": "1",
    "resource_type": "mock host type",
    "vmware_ssh_private_key": "mock vmware SSH key body",
  },
]
`;

exports[`conversion host wizard results step helpers for OSP hosts constructs a POST body for an SSH conversion host without CA certs 1`] = `
Array [
  Object {
    "auth_user": "cloud-user",
    "conversion_host_ssh_private_key": "mock conversion host SSH key body",
    "name": "Mock Host",
    "resource_id": "1",
    "resource_type": "mock host type",
    "vmware_ssh_private_key": "mock vmware SSH key body",
  },
]
`;

exports[`conversion host wizard results step helpers for RHV hosts constructs a POST body for a VDDK conversion host 1`] = `
Array [
  Object {
    "auth_user": "root",
    "conversion_host_ssh_private_key": "mock conversion host SSH key body",
    "name": "Mock Host",
    "resource_id": "1",
    "resource_type": "mock host type",
    "vmware_vddk_package_url": "mock VDDK path",
  },
]
`;

exports[`conversion host wizard results step helpers for RHV hosts constructs a POST body for an SSH conversion host 1`] = `
Array [
  Object {
    "auth_user": "root",
    "conversion_host_ssh_private_key": "mock conversion host SSH key body",
    "name": "Mock Host",
    "resource_id": "1",
    "resource_type": "mock host type",
    "vmware_ssh_private_key": "mock vmware SSH key body",
  },
]
`;
